go-controller/pkg/ovn/obj_retry.go

```diff
@@ -344,6 +357,13 @@ func getResourceKey(objType reflect.Type, obj interface{}) (string, error) {
 			return "", fmt.Errorf("could not cast interface{} object to *kapi.Namespace")
 		}
 		return namespace.Name, nil
+
+	case factory.EgressFirewallType:
+		egressFirewall, ok := obj.(*egressfirewall.EgressFirewall)
+		if !ok {
+			return "", fmt.Errorf("could not cast interface{} object to *egressfirewall.EgressFirewall")
```

Comment:
same thing here, we should print the real Type in the error message. Maybe @ricky-rav can do a follow up to this PR and address this in all the other places as well